### PR TITLE
Replace deprecated _xpack endpoints

### PR DESCRIPTION
**Pull request by ycombinator**
In 7.0.0, Elasticsearch is deprecating most `_xpack/*` endpoints. See https://github.com/elastic/elasticsearch/issues/35958.

This PR updates the Beats codebase, except test fixtures, with the appropriate replacements for the deprecated endpoints.

### CODE REVIEW

**Comment by ycombinator** [comment]
> Will this break compatibility with older versions? I think we should support both paths at least during some versions.

Yes, you are correct. Given that ES is deprecating the old endpoints only starting with 7.0.0, things would break for a user that ran Beats 7.0.0 (with the changes in this PR) against an ES 6.7.0 node. I'm assuming this combination will be supported, following from https://www.elastic.co/support/matrix#matrix_compatibility.

Now, if ES could perform the deprecation in 6.7.0 itself, then the changes in this PR would not cause a break. Note that the changes in this PR will go **only** into 7.0.0/`master` and not be backported.

Do you agree with my assessment above? If so, should I ask on https://github.com/elastic/elasticsearch/issues/35958 if they would be willing to perform the deprecations starting in 6.7.0 itself?

**Comment by ycombinator** [comment]
@jsoriano Any thoughts about my last comment?

**Comment by jsoriano** [comment]
@ycombinator oh sorry, I didn't see this comment.

Yes, I guess that having both paths already in 6.7.0 can also be an option.

**Comment by ycombinator** [comment]
@jsoriano Looks like the new X-Pack APIs used in this PR all exist in ES 6.7.0 already, so there should be no breakage if a user tried to use Beats 7.0.0 (with this PR's changes in it) against ES 6.7.0.

**Comment by ycombinator** [comment]
@ph @kvch @urso Would be good to get one of your 👀 on this PR too, I think. Thanks!

**Comment by jsoriano** [comment]
I'd also prefer something like @urso proposal

**Comment by ycombinator** [comment]
Chatted with @urso off-PR. He explained to me that want to be even more liberal in our compatibility story than what's documented in the support matrix. So I'm going to implement his version check proposal in this PR now.

**Comment by ph** [comment]
@ycombinator do you mind also updating functionbeat's licenser for this change? 
https://github.com/elastic/beats/blob/master/x-pack/functionbeat/licenser/elastic_fetcher.go#L114

**Comment by ycombinator** [comment]
@ph if I'm following that code correctly, it is calling the `GET /_xpack` API. AFAIK, that endpoint is not going away, [even in ES 7.0](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/RestXPackInfoAction.java#L26).

**Comment by ycombinator** [comment]
@urso I've added version checks in various places now, per your recommendation. This PR is ready for another round of review. Thanks!

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ycombinator** [comment]
jenkins, test this

File: metricbeat/module/elasticsearch/ml_job/ml_job.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func init() {
 }
 
 const (
-	jobPath = "/_xpack/ml/anomaly_detectors/_all/_stats"
+	jobPathSuffix = "/anomaly_detectors/_all/_stats"
```
[the remark this answers is not in the document]
Comment (ycombinator) [reply metricbeat/module/elasticsearch/ml_job/ml_job.go]: In 7.0.0+, the preferred path is `_ml/anomaly_detectors/_all/_stats` (notice the underscore in `_ml`). Before that version, the path is `_xpack/ml/anomaly_detectors/_all/_stats` (notice that `ml` does not contain the underscore). Given this difference I decided to only store the **suffix** in the variable here. Further below I concatenate either `/_ml` or `_xpack/ml` to the suffix, depending on the ES version we're talking to.

**Comment by ycombinator** [comment]
> One minor question.

I think the question was about the ML prefix? If so, I've answered it in the comment thread.

> Potentially worth a changelog even though users should not be impacted?

If a user runs a Beat < 7.0.0 with ES >= 7.0.0, they will see deprecation logs in their Elasticsearch logs for Elasticsearch X-Pack API calls made from their Beat. So I added a CHANGELOG entry that says that they won't see such calls with this PR 😄.

> We should add this to our manual testing checklist (if we already have one). I didn't test it locally.

++ where is this checklist kept? I can add this item to it.

**Comment by ycombinator** [comment]
jenkins, test this

**Pull request merged**